vault_core/src/vault.rs

```diff
@@ -827,57 +848,143 @@ impl Vault {
+        if out_amount > max_withdrawable {
             msg!("Amount out exceeds max withdrawable amount");
             return Err(VaultError::VaultUnderflow);
         }
 
         // Slippage check
-        if amount_out < min_amount_out {
+        if out_amount < min_amount_out {
             msg!(
                 "Slippage error, expected more than {} out, got {}",
                 min_amount_out,
-                amount_out
+                out_amount
             );
             return Err(VaultError::SlippageError);
         }
 
         let vrt_supply = self
             .vrt_supply()
-            .checked_sub(amount_to_burn)
+            .checked_sub(burn_amount)
             .ok_or(VaultError::VaultUnderflow)?;
         self.vrt_supply = PodU64::from(vrt_supply);
 
         let tokens_deposited = self
             .tokens_deposited()
-            .checked_sub(amount_out)
+            .checked_sub(out_amount)
             .ok_or(VaultError::VaultUnderflow)?;
         self.tokens_deposited = PodU64::from(tokens_deposited);
 
         Ok(BurnSummary {
             fee_amount,
-            burn_amount: amount_to_burn,
-            out_amount: amount_out,
+            burn_amount,
+            out_amount,
         })
     }
 
+    /// Checks to see if the minimum amount out is acceptable
+    /// acceptable being defined as slippage being larger or equal to the minimum slippage
+    /// as well as accounting for withdraw fees
+    pub fn check_min_supported_mint_out(
+        &self,
+        vrt_amount_in: u64,
+        min_supported_mint_out: u64,
+    ) -> Result<(), VaultError> {
+        if vrt_amount_in == 0 {
+            msg!("Amount in is zero");
+            return Err(VaultError::VaultBurnZero);
+        } else if vrt_amount_in > self.vrt_supply() {
+            msg!("Amount exceeds vault VRT supply");
+            return Err(VaultError::VaultInsufficientFunds);
+        }
+
+        let BurnSummary {
+            fee_amount: _,
+            burn_amount: _,
+            out_amount,
+        } = self.calculate_burn_summary(vrt_amount_in)?;
+
+        let amount_out_delta = out_amount.saturating_sub(min_supported_mint_out);
+        let calculated_slippage = amount_out_delta
+            .checked_mul(MAX_BPS as u64)
+            .and_then(|x| x.checked_div(out_amount))
+            .ok_or(VaultError::VaultOverflow)?;
+
+        if calculated_slippage < Self::MIN_WITHDRAWAL_SLIPPAGE_BPS as u64 {
+            msg!(
+                "Calculated slippage {} is less that the minimum slippage {}",
+                calculated_slippage,
+                Self::MIN_WITHDRAWAL_SLIPPAGE_BPS
+            );
+            return Err(VaultError::SlippageError);
+        }
+
+        // This is a sanity check calculation to ensure that the calculated min amount out is correct
+        let calculated_min_amount_out =
+            self.calculate_min_supported_mint_out(vrt_amount_in, calculated_slippage as u16)?;
```

Comment:
this is doing the same work twice, is that needed?

Reply:
removed